### PR TITLE
Jenkins 56214

### PR DESCRIPTION
**Pull request by TanjaRo1**
Fixed PHP CodeSniffer errors treated as warnings:
Changed severity mapping of CheckStyleParser

### CODE REVIEW

File: src/main/java/edu/hm/hafner/analysis/parser/checkstyle/CheckStyleParser.java [hunk 1]
```diff
@@ -71,8 +71,8 @@ private Report convert(final CheckStyle collection) {
             if (isValidWarning(file)) {
                 for (Error error : file.getErrors()) {
                     IssueBuilder builder = new IssueBuilder();
-                    mapPriority(error).ifPresent(builder::setSeverity);
 
+                    builder.setSeverity(Severity.guessFromString(error.getSeverity()));
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/checkstyle/CheckStyleParser.java]: It is even simpler: `builder.guessSeverity`. I should have noted that in the issue.

File: src/main/java/edu/hm/hafner/analysis/parser/checkstyle/CheckStyleParser.java [hunk 2]
```diff
@@ -71,8 +71,8 @@ private Report convert(final CheckStyle collection) {
...
                     String source = error.getSource();
                     builder.setType(getType(source));
                     builder.setCategory(getCategory(source));
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/checkstyle/CheckStyleParser.java]: The old method `setSeverity` needs to be removed as well.

File: src/test/java/edu/hm/hafner/analysis/parser/CheckStyleParserTest.java [hunk 3]
```diff
@@ -79,6 +79,41 @@ void testParsingOfScalaStyleFormat() {
...
     }
 
+    /**
+     * Test parsing a file and checks the correct servity mapping for error
+     *
+     * @see <a href="https://issues.jenkins-ci.org/browse/JENKINS-56214">Issue 56214</a>
+     */
+    @Test
+    void shouldParseErrorToServityError() {
+        //Read file with one error message
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/CheckStyleParserTest.java]: Do not write comments that describe the code, as these will not survive a refactoring (see [coding style](https://github.com/uhafner/codingstyle)).

Better create a private method and name the method accordingly, e.g. `readReportWithAllSeverities`

File: src/main/java/edu/hm/hafner/analysis/parser/checkstyle/CheckStyleParser.java (hunk 2 above)
Comment (uhafner) [reply src/main/java/edu/hm/hafner/analysis/parser/checkstyle/CheckStyleParser.java]: Sorry: the method that needs to be removed is `mapPriority`

Comment (TanjaRo1) [reply src/main/java/edu/hm/hafner/analysis/parser/checkstyle/CheckStyleParser.java]: Ok sure, already wondered about the duplicated comment.

**Pull request merged**